feat(versioning): allow independent versioning (#984)
algolia/shipjs: website/reference/all-config.md

If `monorepo` is defined, Ship.js will treat the project as a monorepo.

:::warning NOTICE

> Ship.js currently does not provide independent versioning. If you want to use it, you can use the `version` hook and use eg. lerna to update versions.

Reviewer: ```suggestion
Ship.js currently does not support independent versioning out of the box. You can still support it yourself by leveraging the [`version`](#releases) lifecycle hook and use a separate tool (like Lerna) to update versions.
```

1. Ship.js will only publish the packages from `packagesToPublish`.

## `version`

_used at_: `shipjs prepare`

_default_: `undefined`

> If `version` is defined, Ship.js will skip all other validation and creation of changelog. It will only run `version` hook.

Reviewer: Eventually it would be nice to list out exactly what Ship.js usually does for you and _won't_ do if you use `version`, just so you know exactly what you should handle yourself.

```suggestion
When provided, Ship.js skips all other validation and creation of changelog, and only runs `version` instead.
```

Author: ```suggestion
When provided, Ship.js skips all other validation and creation of changelog, and only runs `version` instead. The steps you need to recreate in this hook are:
- decide on the next version (for each package)
- validate the conditions in "shouldPrepare"
- update version numbers
- updating or writing changelog
```
